web3-functions/index.ts

```diff
@@ -0,0 +1,305 @@
+      storedTimestamp: 0,
+    });
+  }
+  conditionalLog(isDebugMode, "Data feed ids: ", dataFeedIds);
+
+  // Wrap contract with redstone data service
+  var wrappedOraclePrimaryProd;
+  switch (dataServiceId) {
+    case "redstone-primary-prod":
+      wrappedOraclePrimaryProd = WrapperBuilder.wrap(oracle).usingDataService(
+        {
+          dataServiceId: "redstone-primary-prod",
+          uniqueSignersCount: 2,
+          dataFeeds: dataFeedIdsString,
+          disablePayloadsDryRun: true,
+        },
+        ["https://oracle-gateway-1.a.redstone.finance"]
+      );
+      break;
+    case "redstone-main-demo":
+      wrappedOraclePrimaryProd = WrapperBuilder.wrap(oracle).usingDataService(
+        {
+          dataServiceId: "redstone-main-demo",
+          uniqueSignersCount: 1,
+          dataFeeds: dataFeedIdsString,
+          disablePayloadsDryRun: true,
+        },
+        ["https://d33trozg86ya9x.cloudfront.net"]
+      );
+      break;
+    default:
+      return {
+        canExec: false,
+        message: `Data service id not found: ${dataServiceId}`,
+      };
+  }
+
+  // Retrieve stored & live prices
+  var { data } =
+    await wrappedOraclePrimaryProd.populateTransaction.getLivePrice(
+      dataFeedIdsBytes32
+    );
+  const txCalldataBytes = arrayify(String(data));
+  const parsingResult = redstone.RedstonePayload.parse(txCalldataBytes);
+
+  conditionalLog(
+    isDebugMode,
+    "Unsigned metadata: ",
+    toUtf8String(parsingResult.unsignedMetadata)
+  );
+  conditionalLog(
+    isDebugMode,
+    "Data packages count: ",
+    parsingResult.signedDataPackages.length
+  );
+  conditionalLog(
+    isDebugMode,
+    "------------------------------------------------------------------------"
+  );
+
+  let dataPackageIndex = 0;
+  for (const signedDataPackage of parsingResult.signedDataPackages) {
+    conditionalLog(
+      isDebugMode,
+      "------------------------------------------------------------------------"
+    );
+    conditionalLog(isDebugMode, `Data package: ${dataPackageIndex}`);
+    conditionalLog(
+      isDebugMode,
+      `Timestamp: ${signedDataPackage.dataPackage.timestampMilliseconds}`
+    );
+    conditionalLog(
+      isDebugMode,
+      `Date and time: ${new Date(
+        signedDataPackage.dataPackage.timestampMilliseconds
+      ).toUTCString()}`
+    );
+    conditionalLog(
+      isDebugMode,
+      "Signer address: ",
+      signedDataPackage.recoverSignerAddress()
+    );
+    conditionalLog(
+      isDebugMode,
+      "Data points count: ",
+      signedDataPackage.dataPackage.dataPoints.length
+    );
+    conditionalLog(
+      isDebugMode,
+      "Data points symbols: ",
+      signedDataPackage.dataPackage.dataPoints.map((dp) => dp.dataFeedId)
+    );
+    conditionalLog(
+      isDebugMode,
+      "Data points values: ",
+      signedDataPackage.dataPackage.dataPoints.map((dp) =>
+        BigNumber.from(dp.value).toNumber()
+      )
+    );
```

Comment:
logging take over big space to follow the logic, maybe we can separate into another func? same for other places

Reply:
Done!